sensor/kubernetes/listener/resources/rbac/store_impl_test.go

```diff
@@ -425,7 +427,128 @@ func TestStore(t *testing.T) {
 	}
+}
+
+func TestStore_DeploymentRelationship(t *testing.T) {
+	t.Setenv("ROX_RESYNC_DISABLED", "true")
+	roles := []*v1.Role{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				UID:       types.UID("r1"),
+				Name:      "r1",
+				Namespace: "n1",
+			},
+			Rules: []v1.PolicyRule{{
+				APIGroups: []string{""},
+				Resources: []string{""},
+				Verbs:     []string{"get"},
+			}},
+		},
+	}
+	clusterRoles := []*v1.ClusterRole{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				UID:  types.UID("r2"),
+				Name: "r2",
+			},
+		},
+	}
+	bindings := []*v1.RoleBinding{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				UID:       types.UID("b1"),
+				Name:      "b1",
+				Namespace: "n1",
+			},
+			RoleRef: v1.RoleRef{
+				Name:     "r1",
+				Kind:     "Role",
+				APIGroup: "rbac.authorization.k8s.io",
+			},
+			Subjects: []v1.Subject{{
+				Kind:      "ServiceAccount",
+				Name:      "sa1",
+				Namespace: "n1",
+			}},
+		},
+	}
+	clusterBindings := []*v1.ClusterRoleBinding{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				UID:  types.UID("b3"),
+				Name: "b3",
+			},
+			RoleRef: v1.RoleRef{
+				Name:     "r2",
+				Kind:     "ClusterRole",
+				APIGroup: "rbac.authorization.k8s.io",
+			},
+			Subjects: []v1.Subject{{
+				Kind:      "ServiceAccount",
+				Name:      "sa2",
+				Namespace: "n1",
+			}},
+		},
+	}
 
+	testCases := map[string]struct {
+		orderedUpdates         []any
+		serviceAccountsUpdates []string
+	}{
+		"Update service account based on single binding": {
+			orderedUpdates:         []any{bindings[0]},
+			serviceAccountsUpdates: []string{"sa1"},
+		},
+		"No service account update if only a role is received": {
+			orderedUpdates:         []any{roles[0]},
+			serviceAccountsUpdates: []string{},
+		},
+		"Update service account both on binding and role update": {
+			orderedUpdates:         []any{bindings[0], roles[0]},
+			serviceAccountsUpdates: []string{"sa1", "sa1"},
+		},
+		"Update service account based on single cluster binding": {
+			orderedUpdates:         []any{clusterBindings[0]},
+			serviceAccountsUpdates: []string{"sa2"},
+		},
+		"No service account update if only a custer role is received": {
+			orderedUpdates:         []any{clusterRoles[0]},
+			serviceAccountsUpdates: []string{},
+		},
+		"Update service account both on cluster binding and cluster role update": {
+			orderedUpdates:         []any{clusterBindings[0], clusterRoles[0]},
+			serviceAccountsUpdates: []string{"sa2", "sa2"},
+		},
+	}
+
+	for name, testCase := range testCases {
+		t.Run(name, func(t *testing.T) {
+			tested := NewStore().(*storeImpl)
+			fakeClient := fake.NewSimpleClientset()
+			dispatcher := NewDispatcher(tested, fakeClient)
+			var ref []resolver.DeploymentReference
+			for _, update := range testCase.orderedUpdates {
+				event := dispatcher.ProcessEvent(update, nil, central.ResourceAction_CREATE_RESOURCE)
+				ref = append(ref, event.DeploymentReference)
+			}
+
+			var orderedServiceAccounts []string
```

Comment:
I feel like `orderedServiceAccounts` is not really used in the test. Maybe you meant to compare it with `testCase.serviceAccountsUpdates`?

Reply:
Oops, you're right. Done